src/main/java/com/jfrog/ide/idea/inspections/JFrogSecurityAnnotator.java

```diff
@@ -37,19 +39,31 @@ public PsiFile collectInformation(@NotNull PsiFile file, @NotNull Editor editor,
 
     @Nullable
     @Override
-    public List<JFrogSecurityWarning> doAnnotate(PsiFile file) {
-        ScanManager scanManager = ScanManagersFactory.getScanManagers(file.getProject()).stream()
-                .findAny()
-                .orElse(null);
-        return scanManager != null ? scanManager.getSourceCodeScanResults() : new ArrayList<>();
+    public List<ApplicableIssueNode> doAnnotate(PsiFile file) {
+        List<ApplicableIssueNode> applicableIssues = new ArrayList<>();
+        ComponentsTree componentsTree = LocalComponentsTree.getInstance(file.getProject());
+        if (componentsTree == null || componentsTree.getModel() == null) {
+            return null;
+        }
+        Enumeration<TreeNode> roots = ((BaseTreeNode) componentsTree.getModel().getRoot()).children();
+        for (TreeNode root : Collections.list(roots)) {
+            FileTreeNode fileNode = (FileTreeNode) root;
+            if (fileNode.getFilePath().equals(file.getContainingFile().getVirtualFile().getPath())) {
+                for (TreeNode issueNode : Collections.list(fileNode.children())) {
+                    applicableIssues.add((ApplicableIssueNode) issueNode);
+                }
+            }
+        }
+        return applicableIssues;
     }
 
     @Override
-    public void apply(@NotNull PsiFile file, List<JFrogSecurityWarning> warnings, @NotNull AnnotationHolder holder) {
-        String filePath = file.getVirtualFile() != null ? file.getVirtualFile().getPath() : null;
-        warnings.stream().filter(Objects::nonNull).filter(warning -> warning.isApplicable() && warning.getFilePath().equals(filePath)).forEach(warning -> {
-            int startOffset = StringUtil.lineColToOffset(file.getText(), warning.getLineStart(), warning.getColStart());
-            int endOffset = StringUtil.lineColToOffset(file.getText(), warning.getLineEnd(), warning.getColEnd());
+    public void apply(@NotNull PsiFile file, List<ApplicableIssueNode> warnings, @NotNull AnnotationHolder holder) {
+        warnings.stream().filter(Objects::nonNull).forEach(warning -> {
```

Comment:
Instead of the `if (document != null)` below
```suggestion
        Document document = PsiDocumentManager.getInstance(file.getProject()).getDocument(file);
        if (document == null) {
            return;
        }
        warnings.stream().filter(Objects::nonNull).forEach(warning -> {
```